tests/framework/InputTemplates/uq_maker.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on February 22 2019
+@author: talbpaul
+
+Template interface for the test UQ template input.
+"""
+from __future__ import print_function, unicode_literals
+import os
+import configparser
+from collections import OrderedDict
+from UQTemplate.UQTemplate import UQTemplate
+
+print('Loading template ...')
+temp = UQTemplate()
+temp.loadTemplate('uq_template.xml', os.path.dirname(__file__))
+print(' ... template loaded')
+
+# information needed by UQTemplate to make run file
+print('Loading input file ...')
+config = configparser.ConfigParser()
+config.read('UQTemplate/uq_template_input.i')
```

Comment:
can we take this from command line?

Reply:
This is up to the developer of an individual template, and seems unnecessary for creating a test and example case.  If you'd like to make this a new requirement for templates, we should start a list of what these requirements are, and figure out a way to enforce template designers to use them.

Reply:
We discussed about this during the meeting. And yes, if there is an input file I think is natural to take it from command line otherwise, if I want to use this Template with a different input I need to hardcode again the file name... 

I am sure you agree with me that hardcoding an input file name is not something we would suggest to anybody out there, so why should we create a test that shows that?

A.

Reply:
That's one application, sure. I don't know that we discussed how specific templates should perscribe reading input files. In this case, the benefit is that it would be very easy to make a desktop shortcut that runs the simulation, so that the user needs only to have the input file open, make changes, and run the shortcut, and there's no need to go to the command line.  So on the contrary, I fully expect one use case of a templated input could be without command line arguments.

The IO you suggest is a good classical standard for *nix-based systems run from the command line; I don't know we need to restrict ourselves to that paradigm.  Of course, if you want to see it that way, I'll change it.

Reply:
The creation of a shortcut with command line arguments is fairly easy. 
In addition, having the hardcoded input file name requires to either specify the full path to it or to run the Driver from the same folder everytime or to create an alias for running it.

You want to keep the hardcoded input file for showing you can do that? Okay... the hardcoded input file name can be the default name and you can create another test where the input file is passed as command line argument.

Okay?